From 1662761837053133fb4402e34bed0393604eab4d Mon Sep 17 00:00:00 2001
From: Peter Fang <peter.fang@intel.com>
Date: Fri, 12 Jun 2020 03:19:07 -0700
Subject: [PATCH 4/5] dm: disable acrntrace to pass build on alpine

Signed-off-by: Peter Fang <peter.fang@intel.com>
---
 misc/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/misc/Makefile b/misc/Makefile
index 90b73a43..845a8477 100644
--- a/misc/Makefile
+++ b/misc/Makefile
@@ -4,7 +4,7 @@ RELEASE ?= 0
 
 .PHONY: all acrn-crashlog acrnlog acrn-manager acrntrace acrnbridge life_mngr
 ifeq ($(RELEASE),0)
-all: acrnlog acrn-manager acrntrace acrnbridge
+all: acrnlog acrn-manager acrnbridge
 else
 all: acrn-manager acrnbridge
 endif
-- 
2.27.0

